packages/airnode-admin/src/cli.ts

```diff
@@ -171,7 +171,7 @@ yargs
     async (args) => {
       const goVerifyAirnodeXpub = goSync(() => admin.verifyAirnodeXpub(args['airnode-xpub'], args['airnode-address']));
       if (!goVerifyAirnodeXpub.success) {
-        logger.error(`Airnode xpub is: INVALID`);
+        logger.log(`Airnode xpub is: INVALID`);
```

Comment:
> The logging works properly only if you called `setLogOptions` within the current execution context 

Hmm, I thought this was the intention (to have the level, timestamp etc.) options enabled for each invocation run :thinking: 

> The logger can be used by deployer as well even before we call the Airnode handlers. It feels to me the logger should be initialized by deployer handlers and [when the Airnode client is invoked](https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/cli/invoke.ts#L7)

Previously the level-specific calls (e.g. `logger.error`) called `logger.consoleLog` instead since no options were passed. The question is that do we now want the options to be available for CLI logs as well (timestamp, level etc.)?

Reply:
> Hmm, I thought this was the intention (to have the level, timestamp etc.) options enabled for each invocation run

Yeah, I just want them to be called when the handler starts, because the handlers code in deployer is run on the cloud provider as well.

> Previously the level-specific calls (e.g. logger.error) called logger.consoleLog instead since no options were passed. The question is that do we now want the options to be available for CLI logs as well (timestamp, level etc.)?

I think it's fine if we use `console.log` internally, but I'd consider that an implementation detail we could change later. (E.g. swap with console.error or use red text via `chalk`). It would be nice to preserve the semantics we have now.

Reply:
Hmmm I see, what changes do you think I should make in this PR and what should be left for later tasks?

Reply:
I'd leave out:
* I wonder if it would be worth it do define CLI specific logger-methods (similar to what airnode-examples has with cliPrint).
* Another thing is that we are mixing "generic purpose logging" with "contextual Airnode logging" (using the pending logs, fixed coordinator ID set at the beginning) which could be split as well.

And implement:
* Set the log options when the handler starts
* Revert https://github.com/api3dao/airnode/pull/1341/commits/35ce00f75323f209a45bd8727c0c3cc03c570ead

Reply:
> Set the log options when the handler starts

I'd need to add config loading to the airnode-node CLI invoke() function to set the level's, right?

Reply:
> I'd need to add config loading to the airnode-node CLI invoke() function to set the levels, right?

That invoke function calls startCoordinator from local handlers so actually probably here https://github.com/api3dao/airnode/blob/renovate/dependencies-(non-major)/packages/airnode-node/src/workers/local-handlers.ts#L17 (This will also ensure the e2e tests are unchanged, since there we call the `startCoordinator` function directly).

> And for this one, I'll set the logOptions inline for the logger call.

Mhh, why not check inside the logging utilities `if (!logOptions) consoleLog(...)` ? That is pretty much the current implementation. (But maybe I am missing something)

Reply:
I don't think we can have that check anymore so that we can call the logger methods and have the `logOptions` be set from logger's state by default even if no `logOptions` are passed in as params (this is why I removed the if-checks) and thought about just using `logger.log` for CLI (with the option of passing logOptions as params). So could we keep the logger.log as the CLI log method, but instead pass in the logOptions as level = ERROR :thinking: .

So with the current changes we can't use logger[level] and conditionally have it pass to `consoleLog` instead :thinking:

Reply:
Thanks for making the effort, I can change it like that :+1:

Reply:
I've unresolved this because Github doesn't handle well links to resolved conversations. (And I plan to link it).